node/test/hyperbahn/rate-limiter.js

```diff
@@ -270,6 +271,7 @@ function runTests(HyperbahnCluster) {
                 send.bind(null, opts),
                 send.bind(null, opts),
                 waitFor(500),
+                send.bind(null, opts),
```

Comment:
@blampe the old behavior is:
1. increment the counters
2. rate limiting check
where all requests are considered for the sake of rate limiting. Therefore, once the rps is beyond the limit, no request can be served.

In the new behavior, it is:
1. rate limiting check. if passes, goto step 2.
2. increment the counters
where only served requests are considered in the counter.